### ISSUE TITLE
Icon launchers

### ISSUE THREAD

**Issue by Metallicow** [wrye-bash/wrye-bash#507]
New Icon launchers.... Feel free to leave comments.
![TaskbarButtonGIF](https://user-images.githubusercontent.com/4668356/80907647-4cf56380-8cde-11ea-93f2-8e5c82732df0.gif)
They cant seem to get the old ones to work right, so a rewrite is in order. Cant have 16x16 icons forever.... These are done in pure python/wxPy

**Comment by Metallicow** [comment]
Ok. managed to get most everything in a basic releasable state for the buttons widget part... I did run into a bit of a hiccup with wx.IconLocation...
Seems that when you feed IconLocation to a wx.Icon, there is not overload for setting preffered icon size, so I had to add in a bit from pywin32 if that is available/OS(I'll have to see if maybe Robin might be able to add an overload for that...). By default it seems that IconLocation either defaults to OS sys ICON_SIZE setting or it is only 16 or 32 size Icons returned for whatever reasons... Bash already handles 16,24,32 sizes, so that kinda makes it moot. Resized Icons will look blurry on HIDPI monitors if 64 or bigger icons are requested unless wx handles that per OS/ContentScalingFactor/Etc....
```python
Icon()
Icon(icon)
Icon(name, type=BITMAP_TYPE_ANY, desiredWidth=-1, desiredHeight=-1)
Icon(loc)
Icon(bmp)
```
On the otherhand most linux systems use png so that is no problem but, *.app may be an issue for mac launchers(Tho afaik nobody has bash running on mac at all yet(ie: retina displays))
The pywin32 code check does work for grabbing(Ex: 256 from like firefox.exe) so as long as it has that icons over 32 wont look pixelated/blownup on windows.

Creating a Custom Widget for the launchers will also need a SetStatusText method or some sort so besides writing a mousewheel scroller for overflow buttons, I think using a splitter setup with either a textctrl or a styledtextctrl if fancy coloring is desired with status text messages.

As far as a container widget, I'm thinking of a control class setup like a notebook tab bar(so as to be scrollable like browser tabs), but with functionality of a agw AUI Toolbar(but without requiring a AuiManager, just a sizer). By doing this, you can eliminate the headache that wx.Statusbar does by force glueing it to the bottom of any main frame(at least on windows). StatusBar/(TaskBar is my name for it) could be used at top or bottom if devs provide the option of preference. 

At the moment the taskbarbutton widget accepts valid paths to exe that would be installed on users systems. This would work out nice with bash.ini with a little work. Bash doesn't use any DLLs for icons for these, so this aspect of the widget doesn't apply here.

**Comment by Infernio** [comment]
Closed in favor of #570.
